include/vara/Feature/Feature.h

```diff
@@ -51,109 +48,15 @@ class Feature {
-    return Excludes.begin();
-  }
-  [[nodiscard]] const_feature_iterator excludes_end() const {
-    return Excludes.end();
-  }
-  llvm::iterator_range<feature_iterator> excludes() {
-    return llvm::make_range(excludes_begin(), excludes_end());
-  }
-  [[nodiscard]] llvm::iterator_range<const_feature_iterator> excludes() const {
-    return llvm::make_range(excludes_begin(), excludes_end());
-  }
-  bool isExcluded(Feature *PosExclude) const {
-    return std::find(excludes_begin(), excludes_end(), PosExclude) !=
-           excludes_end();
-  }
-
-  //===--------------------------------------------------------------------===//
-  // Implications
-
-  feature_iterator implications_begin() { return Implications.begin(); }
-  feature_iterator implications_end() { return Implications.end(); }
-  [[nodiscard]] const_feature_iterator implications_begin() const {
-    return Implications.begin();
-  }
-  [[nodiscard]] const_feature_iterator implications_end() const {
-    return Implications.end();
-  }
-  llvm::iterator_range<feature_iterator> implications() {
-    return llvm::make_range(implications_begin(), implications_end());
-  }
-  [[nodiscard]] llvm::iterator_range<const_feature_iterator>
-  implications() const {
-    return llvm::make_range(implications_begin(), implications_end());
-  }
-  bool isImplied(Feature *PosImplication) const {
-    return std::find(implications_begin(), implications_end(),
-                     PosImplication) != implications_end();
-  }
-
-  //===--------------------------------------------------------------------===//
-  // Alternatives
-
-  feature_iterator alternatives_begin() { return Alternatives.begin(); }
-  feature_iterator alternatives_end() { return Alternatives.end(); }
-  [[nodiscard]] const_feature_iterator alternatives_begin() const {
-    return Alternatives.begin();
-  }
-  [[nodiscard]] const_feature_iterator alternatives_end() const {
-    return Alternatives.end();
-  }
-  llvm::iterator_range<feature_iterator> alternatives() {
-    return llvm::make_range(alternatives_begin(), alternatives_end());
-  }
-  [[nodiscard]] llvm::iterator_range<const_feature_iterator>
-  alternatives() const {
-    return llvm::make_range(alternatives_begin(), alternatives_end());
-  }
-  bool isAlternative(Feature *PosAlternative) const {
-    return std::find(alternatives_begin(), alternatives_end(),
-                     PosAlternative) != alternatives_end();
-  }
-
-  //===--------------------------------------------------------------------===//
-  // Default
-
-  feature_iterator begin() { return children_begin(); }
-  feature_iterator end() { return children_end(); }
-  [[nodiscard]] const_feature_iterator begin() const {
-    return children_begin();
+  [[nodiscard]] Feature *getParentFeature() const {
+    for (auto *P = this->getParent(); P; P = P->getParent()) {
+      auto *F = llvm::dyn_cast<Feature>(P);
+      if (F) {
+        return F;
+      }
+    }
```

Comment:
```suggestion
      if (P->isRoot()) {
        return nullptr;
      }
    }
```
We could add an additional safty check here, not sure if it is worth it. What do you think?

Reply:
Current implementation of `isRoot` just checks `Parent == nullptr`, so one would already expect, that `getParent` on a root node returns a  `nullptr` (which is handled in the loop condition).